app/controllers/service/cypher_controller.rb

```diff
@@ -0,0 +1,24 @@
+# Cypher query via web services API
+
+class Service::CypherController < ServicesController
+
+  def index
+    authorize_user_from_token!
+    if params[:query]
+      # TBD: filter out create, delete, etc
+      cypher = params[:query]
+    else
+      # default for testing.  remove this before release.
+      cypher = "MATCH (n:Page) RETURN n.page_id LIMIT 10;"
+    end
+
+    # Web users are not supposed to modify the database.  The following
+    # is an ad hoc filter to try to prevent this, at least when it is 
+    # accidental; one shouldn't expect it to be secure against a concerted attack.
+    if cypher =~ /\w(delete|create|set|remove|merge|call|drop|load)\w/i
```

Comment:
Errr... I *think* you intended to use `\b` here instead of `\w`. E.g.:

```
DEVELOPMENT [8] pry(main)> cypher = "this has the word create in it"
=> "this has the word create in it"
DEVELOPMENT [9] pry(main)> cypher =~ /\w(delete|create|set|remove|merge|call|drop|load)\w/i
=> nil
DEVELOPMENT [10] pry(main)> cypher =~ /\b(delete|create|set|remove|merge|call|drop|load)\b/i
=> 18
```

Reply:
yep